src/meta/src/barrier/worker.rs

```diff
@@ -571,110 +565,42 @@ impl<C: GlobalBarrierWorkerContext> GlobalBarrierWorker<C> {
                 })?;
             info!(elapsed=?reset_start_time.elapsed(), "control stream reset");
 
-            let mut databases = HashMap::new();
-
             let recovery_result: MetaResult<_> = try {
+                let mut collected_databases = HashMap::new();
+                let mut collecting_databases = HashMap::new();
                 for (database_id, info) in database_fragment_infos {
-                    let source_split_assignments = info
```

Comment:
The deleted code is mostly moved to the `inject_database_initial_barrier` in `src/meta/src/barrier/rpc.rs` with only changing some variable names.